src/metabase/sync/analyze/classify.clj

```diff
@@ -38,9 +38,12 @@
 ;;; |                                         CLASSIFYING INDIVIDUAL FIELDS                                          |
 ;;; +----------------------------------------------------------------------------------------------------------------+
 
-(def ^:private values-that-can-be-set
-  "Columns of Field that classifiers are allowed to set."
-  #{:semantic_type :preview_display :has_field_values :entity_type})
+(defn- updateable-columns
+  "Columns of Field or Table that classifiers are allowed to be set."
+  [model]
+  (case model
+   :model/Field #{:semantic_type :preview_display :has_field_values}
+   :model/Table #{:entity_type}))
```

Comment:
Change looks reasonable, but what's the reason for `case` instead of just a map?

Reply:
I have no preference either way. What's the benefit of a map you see?

Reply:
The case is nice to avoid debugging stray nils. I'd only use a map here if you'd intentionally want to have that fallback, since it's more code and also less efficient.

Reply:
Well, that it kind of seems like configuration rather than execution, which is a map usually :)

Reply:
Also, "less efficient" is funny given that you don't need to execute map and construct those sets every time this function is executed.

Reply:
`case` is actually slightly faster than using a `map`.

```clojure
(defn- updateable-columns
  [model]
  (case model
   :model/Field #{:semantic_type :preview_display :has_field_values}
   :model/Table #{:entity_type}))

(def updateable-columns-2
  {:model/Field #{:semantic_type :preview_display :has_field_values}
   :model/Table #{:entity_type}})

(time (dotimes [_ 100000000]
        (contains? (updateable-columns :model/Table) :entity_type)))
"Elapsed time: 1712.516917 msecs"

(time (dotimes [_ 100000000]
        (contains? (updateable-columns-2 :model/Table) :entity_type)))
"Elapsed time: 1858.564167 msecs" 

(time (dotimes [_ 100000000]
        (contains? (get updateable-columns-2 :model/Table) :entity_type)))
"Elapsed time: 1949.7645 msecs"
```

don't ask me why though

Reply:
wtf 🤣

Reply:
I hadn't thought about the allocations, and shocked to somehow still be right 😂

I was only talking about the choice between mechanisms in the abstract. It's a pity the JIT can't deduce that they are effectively immutable and hoist it #haskellenvy

Reply:
> I was only talking about the choice between mechanisms in the abstract.

What about data over functions for configuration? 😁

Reply:
Damn now you're bringing out the big guns 😂 Can't argue with that